diag_manager/fms_diag_file_object.F90

```diff
@@ -788,6 +824,132 @@ subroutine open_diag_file(this, time_step, file_is_opened)
   diag_file => null()
 end subroutine open_diag_file
 
+!> \brief Write the time metadata to the diag file
+subroutine write_time_metadata(this)
+  class(fmsDiagFileContainer_type), intent(inout), target :: this !< The file object
+
+  class(fmsDiagFile_type), pointer  :: diag_file      !< Diag_file object to open
+  class(FmsNetcdfFile_t),  pointer  :: fileobj        !< The fileobj to write to
+  character(len=50)                 :: time_units_str !< Time units written as a string
+  character(len=50)                 :: calendar       !< The calendar name
+
+  character(len=:), allocatable :: time_var_name !< The name of the time variable as it is defined in the yaml
+
+  diag_file => this%FMS_diag_file
+  fileobj => diag_file%fileobj
+
+  time_var_name = diag_file%get_file_unlimdim()
+  call register_axis(fileobj, time_var_name, unlimited)
+
+  WRITE(time_units_str, 11)  &
+    TRIM(time_unit_list(diag_file%get_file_timeunit())), get_base_year(),&
+         & get_base_month(), get_base_day(), get_base_hour(), get_base_minute(), get_base_second()
+11  FORMAT(a, ' since ', i4.4, '-', i2.2, '-', i2.2, ' ', i2.2, ':', i2.2, ':', i2.2)
+
+  !TODO harcodded "double"
+  call register_field(fileobj, time_var_name, "double", (/time_var_name/))
+  call register_variable_attribute(fileobj, time_var_name, "units", trim(time_units_str), &
+    str_len=len_trim(time_units_str))
+
+  call register_variable_attribute(fileobj, time_var_name, "axis", "T", str_len=1 )
+  call register_variable_attribute(fileobj, time_var_name, "long_name", trim(time_var_name), &
+    str_len=len_trim(time_var_name) )
+
+  !TODO no need to have both attributes, probably?
+  calendar = valid_calendar_types(get_calendar_type())
+  call register_variable_attribute(fileobj, time_var_name, "calendar_type", &
+    uppercase(trim(calendar)), str_len=len_trim(calendar))
+  call register_variable_attribute(fileobj, time_var_name, "calendar", &
+    lowercase(trim(calendar)), str_len=len_trim(calendar))
+
+  if (diag_file%time_ops) call register_variable_attribute(fileobj, time_var_name, "bounds", &
+    trim(time_var_name)//"_bounds", str_len=len_trim(time_var_name//"_bounds"))
+
+end subroutine write_time_metadata
+
+!> \brief Determine if it is time to "write" to the file
+logical function is_time_to_write(this, time_step)
+  class(fmsDiagFileContainer_type), intent(in), target   :: this            !< The file object
+  TYPE(time_type),                  intent(in)           :: time_step       !< Current model step time
+
+  if (time_step >= this%FMS_diag_file%next_output) then
+    is_time_to_write = .true.
+    if (this%FMS_diag_file%is_static) return
+    if (time_step >= this%FMS_diag_file%next_next_output) &
+      call mpp_error(FATAL, this%FMS_diag_file%get_file_fname()//": You skip a time!")
```

Comment:
This error message made me lol, but it is not descriptive enough.  You should at the very least identify it as a diag manager error.